src/Dapper.AOT.Analyzers/CodeAnalysis/DapperInterceptorGenerator.cs

```diff
@@ -1167,24 +1169,59 @@ private static void WriteCommandFactory(string baseFactory, CodeWriter sb, IType
         sb.Outdent().NewLine().NewLine();
     }
 
-    private static void WriteRowFactory(CodeWriter sb, ITypeSymbol type, int index)
+    private static void WriteRowFactory(SourceProductionContext context, CodeWriter sb, ITypeSymbol type, int index)
     {
-        var members = Inspection.GetMembers(type).ToImmutableArray();
-        var memberCount = 0;
-        foreach (var member in members)
+        var members = Inspection.GetMembers(type)
+            .Where(member => CodeWriter.IsSettableInstanceMember(member.Member, out _))
+            .ToImmutableArray();
+        var membersCount = members.Length;
+        if (membersCount == 0)
         {
-            if (CodeWriter.IsSettableInstanceMember(member.Member, out _))
-            {
-                memberCount++;
-            }
+            // report diagnostic? Or do we want to allow returning empty instance like `return new UserType();` ?
```

Comment:
@mgravell, please, review this diagnostic report. I dont think it makes sense to generate "empty" `RowFactory` for the user type, which does not have any settable members (which means that only empty instance can be returned from Dapper).

Let me know if that makes sense, or I need to rollback this change

Reply:
resolving - seems like it's OK